### PR TITLE
picture cards: add person image support

### PR DESCRIPTION
**Pull request by Quentame**
## Proposed change

Add person profile picture support for picture cards, this avoid finding ths profile picture URL in Web Dev Tools, and doing so every time the profile pic is changed/updated.

**Picture Card**
image

**Picture Elements Card**
image

**Picture Entity Card**
image

**Picture Glance Card**
image

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

Already in `gallery/src/pages/lovelace` files

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request: https://github.com/home-assistant/home-assistant.io/pull/32733

## Checklist

- [x] The code change is tested and works locally. (**Tested with Gallery, not as actual HA frontend**)
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

## Summary by CodeRabbit

- **New Features**
  - Introduced support for "person" entities, allowing for the display of person-related data such as user pictures and GPS coordinates in various Lovelace picture cards.

- **UI Enhancements**
  - Updated picture elements, picture entity, and picture glance cards to handle and display person entities.
  - Added new configuration options to support person entities in Lovelace card editors.

- **Configuration**
  - Enhanced Lovelace config to include person entities, improving customization and user experience.

- **Translations**
  - Added new translation entry for "Person entity" to enhance user accessibility.

### CODE REVIEW

File: gallery/public/images/paulus.jpg
Comment (Quentame) [inline gallery/public/images/paulus.jpg:1]: What a great man, and picture 😉

File: src/panels/lovelace/cards/hui-picture-entity-card.ts [hunk 1]
```diff
@@ -108,7 +110,10 @@ class HuiPictureEntityCard extends LitElement implements LovelaceCard {
       return nothing;
     }
 
-    const stateObj = this.hass.states[this._config.entity];
+    const stateObj = this.hass.states[this._config.entity] as
+      | CameraEntity
+      | ImageEntity
+      | PersonEntity;
```
[the remark this answers is not in the document]
Comment (Quentame) [reply src/panels/lovelace/cards/hui-picture-entity-card.ts]: applied on 86b38ba

File: src/data/person.ts [hunk 2]
```diff
@@ -18,6 +22,20 @@ export interface PersonMutableParams {
...
+  editable?: boolean;
+  gps_accuracy?: number;
+  latitude?: number;
+  longitude?: number;
+}
+
+export interface PersonEntity extends HassEntityBase {
+  attributes: PersonEntityAttributes;
+}
+
```
[the remark this answers is not in the document]
Comment (Quentame) [reply src/data/person.ts:38]: yes

**Review by silamon — CHANGES_REQUESTED** [review]
Thank you for the contribution. 👍

Could you refactor `person_entity` into `image_entity` so that `image_entity` can be either a person or a image entity in the picture element card and picture elements card? That would align it with the picture card where `entity` is either a camera, image or person entity and be clear to the user that setting `image_entity` will set the background of the card to the image of the entity regardless if it's a person or image entity.

**Comment by Quentame** [comment]
@silamon `{ name: "image_entity", selector: { entity: { domain: "image" } } },` => Could we set domain to allow only  `image` and `person` domains ?

**Comment by piitaya** [comment]
@Quentame, you can provide multiple domains as a list (https://www.home-assistant.io/docs/blueprint/selectors/#entity-selector). Not sure if it works without the `filter` key.

**Comment by Quentame** [comment]
> @Quentame, you can provide multiple domains as a list (https://www.home-assistant.io/docs/blueprint/selectors/#entity-selector). Not sure if it works without the `filter` key.

OK, thanks !

As my dev env is not ready to test that, I am leaving it as at the moment.

File: src/panels/lovelace/editor/config-elements/hui-picture-card-editor.ts [hunk 3]
```diff
@@ -25,7 +25,7 @@ const cardConfigStruct = assign(
 
 const SCHEMA = [
   { name: "image", selector: { image: {} } },
-  { name: "image_entity", selector: { entity: { domain: "image" } } },
+  { name: "image_entity", selector: { entity: {} } }, // domain: "image" || "person"
```
Comment (silamon) [inline src/panels/lovelace/editor/config-elements/hui-picture-card-editor.ts:28]:
```suggestion
  {
    name: "image_entity",
    selector: { entity: { domain: ["image", "person"] } },
  },
```

**Review by silamon — COMMENTED** [review]
I've tested the code below, should be good to add:

File: src/panels/lovelace/cards/hui-picture-entity-card.ts [hunk 4]
```diff
@@ -136,14 +139,21 @@ class HuiPictureEntityCard extends LitElement implements LovelaceCard {
     }
 
     const domain = computeDomain(this._config.entity);
+    let image: string | undefined = this._config.image;
+    switch (domain) {
+      case "image":
+        image = computeImageUrl(stateObj as ImageEntity);
+        break;
+      case "person":
+        image = (stateObj as PersonEntity).attributes.entity_picture;
```
Comment (silamon) [inline src/panels/lovelace/cards/hui-picture-entity-card.ts:148]: Apparently, picture entity cards are already used for person entities. The default generated lovelace config adds picture entity cards for every known person. If that person entity doesn't have a picture, entity_picture is undefined and it should fallback to the this._config.image. 

```suggestion
        if ((stateObj as PersonEntity).attributes.entity_picture) {
          image = (stateObj as PersonEntity).attributes.entity_picture;
        }
```

Comment (Quentame) [reply src/panels/lovelace/cards/hui-picture-entity-card.ts:148]: Fixed in 349b116e2a7ed9cebfa9f17da01e159b4efb9f16

**Comment by Quentame** [comment]
Doc updated to latest changes: https://github.com/home-assistant/home-assistant.io/pull/32733

**Review by silamon — APPROVED** [review]
LGTM. Thank you for working on this!

**Comment by danielnil** [comment]
@Quentame Is there a way to diable this fuctionality, the person image does not support "moving" GIFs, my picture entity card ussed to be a looping gif, now its just a static image.

**Pull request merged**